servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -418,19 +450,15 @@ protected Hash doOps(
     }
   }
 
-  private static CommitMeta meta(Principal principal, CommitMeta commitMeta)
-      throws NessieConflictException {
-    if (commitMeta.getCommitter() != null) {
-      throw new NessieConflictException(
-          "Cannot set the committer on the client side. It is set by the server.");
-    }
-    String committer = principal == null ? "" : principal.getName();
-    Instant now = Instant.now();
-    return commitMeta.toBuilder()
+  private CommitMeta meta(final CommitMeta referredCommit, Map<String, String> additionalProps) {
+    final String committer = Optional.ofNullable(getPrincipal()).map(p -> p.getName()).orElse("");
+    final Instant now = Instant.now();
+    return referredCommit.toBuilder()
         .committer(committer)
         .commitTime(now)
-        .author(commitMeta.getAuthor() == null ? committer : commitMeta.getAuthor())
-        .authorTime(commitMeta.getAuthorTime() == null ? now : commitMeta.getAuthorTime())
+        .author(referredCommit.getAuthor() == null ? committer : referredCommit.getAuthor())
```

Comment:
Why is this needed with `referredCommit.toBuilder()`?

Reply:
It's existing code, let me know if it is unintended - https://github.com/projectnessie/nessie/blob/c0f2530649c240e89b68a45157bd068c38401178/servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java#L431